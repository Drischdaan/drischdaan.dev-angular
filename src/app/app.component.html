<div class="w-100 h-100 p-5">
  <div class="grid gap-5 grid-cols-4">
    <div class="rounded-lg h-40 bg-cover flex cursor-pointer hover:scale-105 transition-transform duration-300 group" style="background-image: url('https://www.wintotal.de/media/2019/08/software-code-highlighted.jpg');">
      <div class="flex-1 w-100 h-100 bg-gradient-to-t from-neutral-700 rounded-lg p-5 text-white flex flex-col justify-end">
        <h1 class="text-xl font-bold translate-y-5 group-hover:translate-y-0 transition-transform duration-300">Test Project</h1>
        <p class="text-neutral-400 opacity-0 group-hover:opacity-100 transition-opacity duration-300">This is a test project</p>
      </div>
    </div>
  </div>
</div>