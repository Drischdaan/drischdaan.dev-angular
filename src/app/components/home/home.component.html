<div class="w-full h-full flex flex-row">
  <div class="flex-1 flex flex-col justify-center items-center">
    <div class="flex flex-col justify-center items-start gap-5">
      <h1 class="text-5xl font-bold text-purple-400">drischdaan</h1>
      <p class="text-neutral-500">
        A {{ getAge() }} year old software developer from germany. My real name is Tristan,<br>
        but everyone calls me drischdaan.
      </p>
      <div class="flex justify-start items-center gap-3">
        <a href="https://github.com/Drischdaan" target="_blank" class="px-5 py-2 border-2 border-neutral-400 text-neutral-200 hover:bg-neutral-800 transition-colors duration-300">
          Github
        </a>
        <a href="https://www.instagram.com/drischdaan/" target="_blank" class="px-5 py-2 border-2 border-neutral-400 text-neutral-200 hover:bg-neutral-800 transition-colors duration-300">
          Instagram
        </a>
        <a href="https://open.spotify.com/user/wqqh06uhneiz0mm94aqsgzhte?si=a5510b1ec5914c92" target="_blank" class="px-5 py-2 border-2 border-neutral-400 text-neutral-200 hover:bg-neutral-800 transition-colors duration-300">
          Spotify
        </a>
      </div>
    </div>
  </div>
  <div class="flex-1 flex flex-col justify-center items-center gap-5">
    <img src="assets/images/temp-picture.png" width="300">
    <p class="italic text-neutral-600">*This is temporary*</p>
  </div>
</div>