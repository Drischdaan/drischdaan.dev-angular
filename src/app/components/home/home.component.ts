import { NgtGLTFLoader } from '@angular-three/soba/loaders';
import { Component } from '@angular/core';
import { AnimationClip, Color, Group, Mesh } from 'three';

@Component({
  selector: 'app-home',
  templateUrl: './home.component.html',
  styleUrls: ['./home.component.scss'],
})
export class HomeComponent {

  color: Color = new Color('#0c0b0c');
  monitor$ = this.loaderService.load('assets/models/Monitor.glb');

  constructor(
    private readonly loaderService: NgtGLTFLoader,
  ) { }

  public getAge(): number {
    const today: Date = new Date();
    const birthDate: Date = new Date('2003/06/23'); // Leaked my birthday lol
    let age: number = today.getFullYear() - birthDate.getFullYear();
    const month: number = today.getMonth() - birthDate.getMonth();
    if (month < 0 || (month === 0 && today.getDate() < birthDate.getDate())) {
      age--;
    }
    return age;
  }

  onReady(model: Group) {
    model.traverse((child) => {
      if ((child as Mesh).isMesh) {
        child.castShadow = true;
        child.receiveShadow = true;
      }
    });
  }

}