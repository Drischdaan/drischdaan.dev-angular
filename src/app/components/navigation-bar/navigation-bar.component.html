<ng-container>
  <div class="absolute top-0 left-0 w-full p-3 grid grid-cols-3 align-middle text-white">
    <div class="flex justify-start items-center">
      <h1 class="text-xl font-bold text-transparent bg-clip-text bg-gradient-to-r from-purple-400 to-pink-600">drischdaan</h1>
    </div>
    <div class="hidden sm:flex justify-center items-center">
      <ul class="flex justify-center items-center gap-3">
        <li *ngFor="let item of items">
          <button
            (click)="onNavigationItemClicked(item.id)"
            class="hover:text-purple-400 hover:scale-105 transition-colors duration-300"
          >
            {{ item.name }}
          </button>
        </li>
      </ul>
    </div>
    <div class="block sm:hidden"></div>
    <div class="sm:hidden flex justify-end items-center">
      <button
        (click)="onMobileMenuClicked(true)"
        class="px-5 py-2 border-2 border-transparent hover:border-white hover:bg-neutral-700 transition-colors duration-300"
      >
        Menu
      </button>
    </div>
  </div>
  <div *ngIf="mobileMenuOpen$ | async" class="fixed top-0 left-0 w-screen h-screen bg-neutral-900 text-white">
    <div class="fixed w-full top-0 left-0 grid grid-cols-3 p-5">
      <div class="flex justify-start items-center">
        <button
          (click)="onMobileMenuClicked(false)"
          class="px-5 py-2 border-2 border-transparent hover:border-white hover:bg-neutral-700 transition-colors duration-300"
        >
          Exit
        </button>
      </div>
      <div class="flex justify-center items-center">
        <h1 class="text-xl font-bold text-transparent bg-clip-text bg-gradient-to-r from-purple-400 to-pink-600">drischdaan</h1>
      </div>
    </div>
    <ul class="h-screen flex justify-center items-center flex-col gap-2">
      <li *ngFor="let item of items" class="w-full px-5 flex justify-center items-center">
        <button
          (click)="onNavigationItemClicked(item.id); onMobileMenuClicked(false)"
          class="w-full py-5 border-2 border-transparent hover:border-white hover:bg-neutral-700 transition-colors duration-300"
        >{{ item.name }}</button>
      </li>
    </ul>
  </div>
</ng-container>