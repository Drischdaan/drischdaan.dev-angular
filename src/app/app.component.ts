import { Component } from '@angular/core';
import { INavigationItem } from './components/navigation-bar/models/navigation-bar.models';

@Component({
  selector: 'app-root',
  templateUrl: './app.component.html',
  styleUrls: ['./app.component.scss']
})
export class AppComponent {

  navigationItems: INavigationItem[] = [
    {
      id: 'home',
      name: 'Home',
    },
    {
      id: 'about',
      name: 'About',
    },
    {
      id: 'projects',
      name: 'Projects',
    },
    {
      id: 'contact',
      name: 'Contact',
    },
  ];

}