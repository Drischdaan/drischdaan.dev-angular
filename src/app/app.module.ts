import { NgModule } from '@angular/core';
import { BrowserModule } from '@angular/platform-browser';

import { AppComponent } from './app.component';
import { ProjectCardComponent } from './components/project-card/project-card.component';
import { NavigationBarComponent } from './components/navigation-bar/navigation-bar.component';

@NgModule({
  declarations: [
    AppComponent,
    ProjectCardComponent,
    NavigationBarComponent
  ],
  imports: [
    BrowserModule
  ],
  providers: [],
  bootstrap: [AppComponent]
})
export class AppModule { }