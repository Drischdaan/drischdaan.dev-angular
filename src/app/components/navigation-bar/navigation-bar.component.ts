import { Component, Input } from '@angular/core';
import { Router } from '@angular/router';
import { BehaviorSubject } from 'rxjs';
import { INavigationItem } from './models/navigation-bar.models';

@Component({
  selector: 'app-navigation-bar',
  templateUrl: './navigation-bar.component.html',
  styleUrls: ['./navigation-bar.component.scss']
})
export class NavigationBarComponent {

  @Input() items: INavigationItem[] = [];

  mobileMenuOpen$: BehaviorSubject<boolean> = new BehaviorSubject(true);

  constructor(
    private readonly router: Router
  ) { }

  public onMobileMenuClicked(value: boolean): void {
    this.mobileMenuOpen$.next(value);
  }

  public onNavigationItemClicked(id: string): void {
    this.router.onSameUrlNavigation = "reload";
    this.router.navigate(['/'], { fragment: id }).finally(() => {
      this.router.onSameUrlNavigation = "ignore";
    });
  }

}
