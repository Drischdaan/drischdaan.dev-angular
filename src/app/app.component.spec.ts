import { ComponentFixture, TestBed } from '@angular/core/testing';
import { AppComponent } from './app.component';
import { HomeComponent } from './components/home/home.component';
import { NavigationBarComponent } from './components/navigation-bar/navigation-bar.component';

describe('AppComponent', () => {
  let fixture: ComponentFixture<AppComponent>;
  let component: AppComponent;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      declarations: [
        AppComponent,
        NavigationBarComponent,
        HomeComponent
      ],
    }).compileComponents();

    fixture = TestBed.createComponent(AppComponent);
    component = fixture.componentInstance;
  });

  it('should create the component', () => {
    expect(component).toBeTruthy();
  });

  it('should match snapshot', () => {
    expect(component).toMatchSnapshot();
  });
});
