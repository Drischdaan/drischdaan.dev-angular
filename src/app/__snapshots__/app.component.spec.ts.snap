// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppComponent should match snapshot 1`] = `
AppComponent {
  "__ngContext__": 3,
  "navigationItems": [
    {
      "id": "home",
      "name": "Home",
    },
    {
      "id": "about",
      "name": "About",
    },
    {
      "id": "projects",
      "name": "Projects",
    },
    {
      "id": "contact",
      "name": "Contact",
    },
  ],
}
`;
